samples/cpp/houghcircles.cpp

```diff
@@ -24,39 +24,47 @@ int main(int argc, char** argv)
         help();
         return 0;
     }
+    //![load]
     string filename = parser.get<string>("@image");
-    if (filename.empty())
-    {
-        help();
-        cout << "no image_name provided" << endl;
-        return -1;
-    }
-    Mat img = imread(filename, 0);
+    Mat img = imread(filename, IMREAD_COLOR);
     if(img.empty())
     {
         help();
         cout << "can not open " << filename << endl;
         return -1;
     }
+    //![load]
+
+    //![convert_to_gray]
+    Mat gray;
+    cvtColor(img, gray, COLOR_BGR2GRAY);
+    //![convert_to_gray]
 
-    Mat cimg;
-    medianBlur(img, img, 5);
-    cvtColor(img, cimg, COLOR_GRAY2BGR);
+    //![reduce_noise]
+    medianBlur(gray, gray, 5);
+    //![reduce_noise]
 
+    //![houghcircles]
     vector<Vec3f> circles;
-    HoughCircles(img, circles, HOUGH_GRADIENT, 1, 10,
+    HoughCircles(gray, circles, HOUGH_GRADIENT, 1, gray.rows/16,
```

Comment:
Is there a reason why you are chosing rows/16 here as the default value? It will depend on the application and thus maybe a remark above should point this out?